bftengine/src/bftengine/ReplicaImp.hpp

```diff
@@ -625,6 +630,10 @@ class ReplicaImp : public InternalReplicaApi, public ReplicaForStateTransfer {
   void onCarrierMessage(CarrierMesssage* msg);
 
  private:
+  // vector holding "stop" functions of components dependent on replicaImp that need to stop before the
+  // replica is stopped
+  std::vector<std::function<void(void)>> stopFuncs_;
```

Comment:
since currently it's only for the preprocessor and we don't know what will be needed in the future I prefer keeping it simple until we know otherwise.

Reply:
done